### PR TITLE
feat: ensure webpack is built before e2e tests

### PR DESCRIPTION
**Pull request by HussainTaj-arbisoft**
#### Pre-Flight checklist

- [x] Testing
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Closes https://github.com/mitodl/ocw-hugo-themes/issues/1140

#### What's this PR do?
This PR:

- Updates webpack.dev.ts config to enable the dev server that writes to disk.
- Adds a command `yarn webpack:serve` to run the dev server.
- Adds a bash script that checks for the webpack server. If the server is not running, builds webpack. If the server is running, skips building webpack. And runs e2e tests regardless. The server checks only happen in dev mode.
- Updates `yarn test:e2e` to run the new bash script.
- Removes the `Build Webpack` task from CI. Since`yarn test:e2e` always runs in dev mode, a build is ensured by it. So the CI task is unnecessary.

#### Where should the reviewer start?
Best to start from the core change `package_scripts/test_e2e.sh`

#### How should this be manually tested?

We'll start off by testing the existing commands to make sure they work the same as before. Then we'll proceed to test the new commands/features.

1. Navigate to your local ocw-hugo-themes setup.
2. Checkout branch `hussaintaj/1140-webpack-e2e`
3. Run `yarn build:webpack`.
    **Expected Behavior**: Build is successful. Depending on the value of `WEBPACK_WATCH_MODE` env var, this command either finishes after the build or watches changes for continuous builds.
4. Run `yarn build:webpack:dev`.
    **Expected Behavior**: Build is successful.
5. Run `yarn test:e2e`
    **Expected Behavior**: All tests pass. You see output similar to:
    ```log
    Webpack server at localhost:3001 is not responsive.
    Run webpack server in the background for faster start-up times.

             yarn webpack:serve 

    Building webpack...

    <wepback logs>
    <build and test logs>
    ```
6. Run `yarn webpack:serve` in a separate terminal. And run `yarn test:e2e`.
    **Expected Behavior**: All tests pass. You see output similar to:
    ```log
    Connection to localhost port 3001 [tcp/redwood-broker] succeeded!
    Found Webpack server at localhost:3001, skipping manual build.
    <build and test logs>
    ```
7. While keeping the webpack server running, make a change that would trigger a build and make a test fail. For example, you could remove this line https://github.com/mitodl/ocw-hugo-themes/blob/140f31c507222157c973532536b426594f345122/course-v2/assets/course-v2.ts#L31
    **Expected Behavoir**: A few tests fail.
8. Undo your code changes in step 7. Run `yarn test:e2e video-tabs.spec.ts`.
    **Expected Behavior**: Only the specified test module is run.

#### Additional Information

In the testing section step 7 above, you'll be making a change to some code to trigger a build. During my testing, I realized that changing dynamic imports were glitchy (not always hot reloaded by the dev server). The closest reported issue I found to this was https://github.com/webpack/webpack-dev-server/issues/2792. It might be a configuration issue or a dev-server issue, I'm not sure. But it doesn't directly relate to this PR's ticket. I might do some more digging later to find out the cause. In the meantime, I believe we can proceed with this configuration.

### CODE REVIEW

File: package_scripts/test_e2e.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        if ! yarn build:webpack:dev; then
+            echo -e "\n${RED}Command 'yarn build:webpack:dev' failed. Exiting...${RESET}\n"
+            exit 1
+        fi
+    else
+        echo -e "${GREEN}Found Webpack server at $WEBPACK_HOST:$WEBPACK_PORT, skipping manual build.${RESET}"
+    fi
+fi
+
+playwright test $@
```
Comment (ChristopherChudzicki) [inline package_scripts/test_e2e.sh:27]: I'm not great with bash scripts. Now I know about `$@`. Thanks!

**Review by ChristopherChudzicki — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -15,9 +15,10 @@
     "build": "yarn with-env -- ./package_scripts/build.sh",
     "build:githash": "mkdir -p base-theme/dist/static && git rev-parse HEAD > base-theme/dist/static/hash.txt",
     "build:hugo": "hugo -d ../dist -v",
-    "build:webpack:dev": "cross-env NODE_ENV=development WEBPACK_WATCH_MODE=false webpack --config base-theme/assets/webpack/webpack.dev.ts",
     "build:hugo:debug": "hugo -d ../dist -v --templateMetrics --debug",
     "build:webpack": "cross-env NODE_ENV=production webpack --config base-theme/assets/webpack/webpack.prod.ts",
+    "build:webpack:dev": "cross-env NODE_ENV=development WEBPACK_WATCH_MODE=false webpack --config base-theme/assets/webpack/webpack.dev.ts",
+    "webpack:serve": "cross-env NODE_ENV=development webpack serve --config base-theme/assets/webpack/webpack.dev.ts",
```
[the remark this answers is not in the document]
Comment (HussainTaj-arbisoft) [reply package.json:21]: Thank you for the suggestion. Doesn't make sense to have two commands doing the same thing. I'll work on it.

[the remark this answers is not in the document]
Comment (HussainTaj-arbisoft) [reply package.json:21]: @ChristopherChudzicki I worked on switching to the server for local development but the problem with dynamic imports appeared again. Dynamic imports do not get hot reloaded correctly. I investigated and was led to the same conclusion (possibly https://github.com/webpack/webpack-dev-server/issues/2792). We likely need to update our Webpack configurations.

The dynamic imports' assets update works correctly in watch mode builds. So for now, I believe keeping the watch mode for development is better.

I'll create a ticket for this issue shortly.

[the remark this answers is not in the document]
Comment (HussainTaj-arbisoft) [reply package.json:21]: https://github.com/mitodl/ocw-hugo-themes/issues/1172

**Pull request merged**